geode/src/main/scala/akka/stream/alpakka/geode/GeodeSettings.scala

```diff
@@ -13,23 +13,24 @@ import org.apache.geode.cache.client.ClientCacheFactory
  * @param port      default to 10334
  * @param pdxCompat a function that determines if two class are equivalent (java class / scala case class)
  */
-case class GeodeSettings(hostname: String,
-                         port: Int = 10334,
-                         configure: Option[ClientCacheFactory => ClientCacheFactory] = None,
-                         pdxCompat: (Class[_], Class[_]) => Boolean = (c1, c2) =>
-                           c1.getSimpleName equals c2.getSimpleName) {
+class GeodeSettings(val hostname: String,
+                    val port: Int = 10334,
+                    val configure: Option[ClientCacheFactory => ClientCacheFactory] = None,
+                    val pdxCompat: (Class[_], Class[_]) => Boolean = (c1, c2) =>
+                      c1.getSimpleName equals c2.getSimpleName) {
 
   /**
    * @param configure function to configure geode client factory
    */
   def withConfiguration(configure: ClientCacheFactory => ClientCacheFactory) =
-    copy(configure = Some(configure))
+    new GeodeSettings(hostname, port, Some(configure), pdxCompat)
```

Comment:
You may implement a private `copy` method. Have a look at the new reference connector, it shows good patterns for settings classes.